os/test/src/PathTests.scala

```diff
@@ -2,16 +2,90 @@ package test.os
 
 import java.nio.file.Paths
 import java.io.File
-
 import os._
-import os.Path.{driveRoot}
+import os.Path.driveRoot
+import os.PathChunk.segmentsFromString
 import utest.{assert => _, _}
+
 import java.net.URI
 object PathTests extends TestSuite {
+  private def nonValidPathSegment(chars: String) = s"[$chars] is not a valid path segment."
+
   val tests = Tests {
+    test("segmentsFromString") {
+      def testSegmentsFromString(s: String, expected: List[String]) = {
+        assert(segmentsFromString(s).sameElements(expected))
+      }
+
+      testSegmentsFromString("  ", "  " :: Nil)
+
+      testSegmentsFromString("", "" :: Nil)
+
+      testSegmentsFromString("""foo/bar/baz""", "foo" :: "bar" :: "baz" :: Nil)
+
+      testSegmentsFromString("""/""", "" :: "" :: Nil)
+      testSegmentsFromString("""//""", "" :: "" :: "" :: Nil)
+      testSegmentsFromString("""///""", "" :: "" :: "" :: "" :: Nil)
+
+      testSegmentsFromString("""a/""", "a" :: "" :: Nil)
+      testSegmentsFromString("""a//""", "a" :: "" :: "" :: Nil)
+      testSegmentsFromString("""a///""", "a" :: "" :: "" :: "" :: Nil)
+
+      testSegmentsFromString("""ahs/""", "ahs" :: "" :: Nil)
+      testSegmentsFromString("""ahs//""", "ahs" :: "" :: "" :: Nil)
+
+      testSegmentsFromString("""ahs/aa/""", "ahs" :: "aa" :: "" :: Nil)
+      testSegmentsFromString("""ahs/aa//""", "ahs" :: "aa" :: "" :: "" :: Nil)
+
+      testSegmentsFromString("""/a""", "" :: "a" :: Nil)
+      testSegmentsFromString("""//a""", "" :: "" :: "a" :: Nil)
+      testSegmentsFromString("""//a/""", "" :: "" :: "a" :: "" :: Nil)
```

Comment:
Let's use the `List(...)` syntax here rather than `::` and `Nil`, for better consistency with the rest of the codebase

Reply:
Actually I am thinking of making this function private[os] , in order not to expose it to public API. 

This means dropping tests, but keeping it public will force us to keep binary compatibility, and can result in users using it to form a paths (which we probably dont want)

Reply:
can we move the tests into the same package? I know the tests are mosty in a different package for now, but i dont think that's required